js/token-tree.js

```diff
@@ -36,9 +47,35 @@
    */
   function selectEventHandler (event) {
     if (!event.choice.children) {
+      insertTokenIntoTextBox.call(this, event);
+
       return;
     }
+    toggleTreeElement(event);
+  }
+
+  /**
+   * @param {object} event event object
+   */
+  function insertTokenIntoTextBox (event) {
+    var token = event.choice.id;
+    var field = $(this).data('field');
+    if (field.indexOf('html') < 0) {
+      field = textMsgID($(this));
+    }
+    CRM.wysiwyg.insert('#' + field, token);
+    $(this).select2('val', '');
+    if (isMailing) {
+      verify();
```

Comment:
what does `verify()` do? Can we use a longer name that explains what's going on in a better way?

Reply:
This function is defined in core. so cant change it.